File: app/src/main/java/n7/mev/main/MainViewModelFactory.java
package n7.mev.main;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class MainViewModelFactory extends ViewModelProvider.NewInstanceFactory {

    private final String moduleName;
    private final Application application;

    public MainViewModelFactory(Application application, String moduleName) {
        this.moduleName = moduleName;
        this.application = application;
    }

    @SuppressWarnings("unchecked")
    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        if (modelClass == MainViewModel.class) {
            return (T) new MainViewModel(application,moduleName);
        }
        return super.create(modelClass);
    }
}

File: app/src/main/java/n7/mev/main/MainViewModelFactory.kt
package n7.mev.main

import android.app.Application
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider.NewInstanceFactory

class MainViewModelFactory(private val application: Application, private val moduleName: String) : NewInstanceFactory() {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return if (modelClass == MainViewModel::class.java) {
            MainViewModel(application, moduleName) as T
        } else super.create(modelClass)
    }

}